bluemira/base/parameter_frame/_frame.py

```diff
@@ -115,14 +186,133 @@ def to_dict(self) -> Dict[str, Dict[str, Any]]:
     @classmethod
     def _validate_parameter_field(cls, field: str) -> Tuple[Type, ...]:
         member_type = cls.__dataclass_fields__[field].type
-        if (member_type is not NewParameter) and (
+        if (member_type is not Parameter) and (
             not hasattr(member_type, "__origin__")
-            or member_type.__origin__ is not NewParameter
+            or member_type.__origin__ is not Parameter
         ):
-            raise TypeError(f"Field '{field}' does not have type NewParameter.")
+            raise TypeError(f"Field '{field}' does not have type Parameter.")
         value_types = get_args(member_type)
         return value_types
 
+    @classmethod
+    def _validate_units(cls, param_data: Dict, value_type):
+
+        quantity = pint.Quantity(param_data["value"], param_data["unit"])
+
+        if dimensionality := quantity.units.dimensionality:
+            unit = cls._fix_combined_units(cls._remake_units(dimensionality))
+        else:
+            unit = quantity.units
+
+        unit = cls._fix_weird_units(unit, quantity.units)
+
+        if unit != quantity.units:
+            val = raw_uc(quantity.magnitude, quantity.units, unit)
+            if isinstance(param_data["value"], int) and int in value_type:
+                val = int(val)
+            param_data["value"] = val
+
+        param_data["unit"] = unit
+
+    @staticmethod
```

Comment:
This is probably more of a personal preference, but I would usually make a private static method a free function. There's no functional difference, but it doesn't expose the methods to users of the class and makes the class definition shorter and cleaner.